server/src/main/java/org/opensearch/action/search/SearchRequest.java

```diff
@@ -678,6 +679,17 @@ public SearchTask createTask(long id, String type, String action, TaskId parentT
         return new SearchTask(id, type, action, this::buildDescription, parentTaskId, headers, cancelAfterTimeInterval);
     }
 
+    public SearchTask createTask(
+        long id,
+        String type,
+        String action,
+        Supplier<String> description,
+        TaskId parentTaskId,
+        Map<String, String> headers
+    ) {
+        return new SearchTask(id, type, action, description, parentTaskId, headers, cancelAfterTimeInterval);
+    }
+
```

Comment:
Why is this needed? Does the default build description not work?

Reply:
it'll work but the description will be of search request and here we're passing the custom description of create pit from create pit request. if we are okay with default search description, then i can just use the default search task.